### PR TITLE
Add new developer guide

### PR DESCRIPTION
**Pull request by wanliAlex**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Docs update

* **What is the current behavior?** (You can also link to an open issue here)
the developer guide is outdated

* **What is the new behavior (if this is a feature change)?**
it works with marqo v2

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
no

* **Related Python client changes** (link commit/PR here)
no

* **Related documentation changes** (link commit/PR here)
no

* **Other information**:
no

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: src/marqo/README.md [hunk 1]
```diff
@@ -4,47 +4,88 @@ you want.
...
 ```bash
-docker run --name marqo-os -id -p 9200:9200 -p 9600:9600 -e "discovery.type=single-node" marqoai/marqo-os:0.0.3
+git clone https://github.com/marqo-ai/marqo.git
+cd marqo
 ```
 
-2. Clone the github repo
+2. Install Marqo dependencies
+```bash
+pip install -r requirements.txt
```
Comment (danyilq) [inline src/marqo/README.md]: just curious, isn't it a place where we need to recommend requirements.dev.txt?

File: src/marqo/README.md [hunk 2]
```diff
@@ -4,47 +4,88 @@ you want.
...
+3. Pull and run the Vespa docker image
+```bash
+docker run --detach --name vespa --hostname vespa-tutorial \
+  --publish 8080:8080 --publish 19071:19071 --publish 19092:19092 \
+  vespaengine/vespa:latest
 ```
-3. Install marqo dependencies
+
+4. Install the vespa command line interface (CLI)
+Check out the [Vespa CLI documentation](https://docs.vespa.ai/en/vespa-cli.html) for how to install the CLI.
```
Comment (danyilq) [inline src/marqo/README.md]: I felt really strong urge to skip this step as it looks not really meaningful when you are lazy, I think experience would be just a little bit better if we include that stroke for installation there and point it to the documentation as well.

File: src/marqo/README.md (hunk 1 above)
Comment (wanliAlex) [reply src/marqo/README.md]: Thanks for getting this. Let me update it

File: src/marqo/README.md (hunk 2 above)
Comment (wanliAlex) [reply src/marqo/README.md]: I update the readme so we use curl. ez and simple now

**Review by danyilq — APPROVED** [review]
lgtm

**Pull request merged**